src/edu/stanford/nlp/optimization/QNMinimizer.java

```diff
@@ -328,7 +328,6 @@ public MaxEvaluationsExceeded(String s) {
     private int maxSize = 100; // This will control the number of func values /
     // gradients to retain.
     private Function mon = null;
-    private boolean quiet = false;
```

Comment:
is this is a case of things having diverged from the time you first wrote it?  below, all the debug messages are calls to sayln()

Reply:
No, this IMHO is even a subtle bug. There are two `quiet` accessible in this context.
https://github.com/stanfordnlp/CoreNLP/blob/eb43d5d9150de97f8061fa06b838f1d021586789/src/edu/stanford/nlp/optimization/QNMinimizer.java#L103
https://github.com/stanfordnlp/CoreNLP/blob/eb43d5d9150de97f8061fa06b838f1d021586789/src/edu/stanford/nlp/optimization/QNMinimizer.java#L331
My patch intentionally removes `sayln` because that is an anti-pattern in logging. As is, it would first construct the log message, then call sayln, and only then notice that it is silenced. The "quiet" check should be outside; redundant checks can then be merged.

Reply:
I had missed the first quiet.  Thanks for pointing that out.